tokio-stream/src/lib.rs

```diff
@@ -64,9 +64,9 @@
 //! types when the io feature is enabled.
 //!
 //! [`tokio-util`]: https://docs.rs/tokio-util/0.4/tokio_util/codec/index.html
-//! [`tokio::io`]: https://docs.rs/tokio/1.0/tokio/io/index.html
-//! [`AsyncRead`]: https://docs.rs/tokio/1.0/tokio/io/trait.AsyncRead.html
-//! [`AsyncWrite`]: https://docs.rs/tokio/1.0/tokio/io/trait.AsyncWrite.html
+//! [`tokio::io`]: https://docs.rs/tokio/1.17.0/tokio/io/index.html
+//! [`AsyncRead`]: https://docs.rs/tokio/1.17.0/tokio/io/trait.AsyncRead.html
+//! [`AsyncWrite`]: https://docs.rs/tokio/1.17.0/tokio/io/trait.AsyncWrite.html
```

Comment:
I wonder if this should just be using `latest` rather than a specific version?

Reply:
I think that changing this might cause problems for IDEs that provide links to the docs when you hover over an item, as if a user was working with an older version of the library, the link might confusingly take them to the docs for the latest version if the IDE used this string to get the root url for the docs.

Reply:
Oh nvm I see. I mistakenly thought we were referring to the one in Cargo.toml. Yeah I actually agree with you here, this probably should link to the latest.

It would be nice if we could get links that were semver aware, i.e. 1.X releases would have a dedicated latest so that API changes in a 2.X release wouldn't break older docs though.

Reply:
most READMEs and stuff are using `latest`; but the point Noah makes is also valid. it looks like we haven't been very good at keeping these up to date, even though there's a script for doing it, maybe we're better off switching to `latest`? idk.

i'm also happy to just back those changes out of this PR.

Reply:
> It would be nice if we could get links that were semver aware, i.e. 1.X releases would have a dedicated latest so that API changes in a 2.X release wouldn't break older docs though.

i believe you can do that: https://docs.rs/tokio/1/tokio should take you to the latest v1.x.y version.

i can change these to do that in a separate branch?

Reply:
...oh, huh, https://docs.rs/tokio/1.0/tokio _also_ links to 1.16.1. that's...not what i would have expected! but, i guess it means these are actually fine as-is.

Reply:
I did not realize that we could do this! Yeah, I it would be good to open an issue and separate PR to change all cases where we do this to point to the 1.X latest, but that shouldn't block this release.

Reply:
@hawkw apparently these strings are semver.

Reply:
I'm sitting right next to one of the docs.rs maintainers so I asked him :laughing: